File: Codes/3379-transformed-array.go
/*
 * @Author: LetMeFly
 * @Date: 2026-02-15 18:09:49
 * @LastEditors: LetMeFly.xyz
 * @LastEditTime: 2026-02-15 18:15:07
 */
package main

func constructTransformedArray(nums []int) []int {
    ans := make([]int, len(nums))
    n := len(nums)
    for i, v := range nums {
        ans[i] = nums[((v + i) % n + n) % n]
    }
    return ans
}

File: Codes/3379-transformed-array.cpp
/*
 * @Author: LetMeFly
 * @Date: 2026-02-15 18:09:49
 * @LastEditors: LetMeFly.xyz
 * @LastEditTime: 2026-02-15 18:11:29
 */
#if defined(_WIN32) || defined(__APPLE__)
#include "_[1,2]toVector.h"
#endif

class Solution {
public:
    vector<int> constructTransformedArray(vector<int>& nums) {
        vector<int> ans(nums.size());
        for (int i = 0, n = nums.size(); i < n; i++) {
            ans[i] = nums[((i + nums[i]) % n + n) % n];
        }
        return ans;
    }
};